File: twosum.py
def twosum(arr, target):
    temp = dict()

    for i in arr:
        temp[i] = 1
    
    for i in arr:
        if (target - i) in temp:
            return [i, target-i]
    
    return 'Not Available'
    


if __name__ == "__main__":

    arr = [4,5,6,7,89]
    target = 13

    print(twosum(arr, target))

File: TwoSum.java
import java.util.*;

public class TwoSum {

    public int[] twoSum(int[] arr, int target) {

        int[] answer = new int[2];

        HashMap<Integer, Integer> hm = new HashMap<>();

        for (int index = 0; index < arr.length; index++) {

            hm.put(arr[index], index);
        }

        System.out.println(hm);

        for (int index = 0; index < arr.length; index++) {

            if (hm.containsKey(target - arr[index])) {

                answer = new int[] { index, hm.get(target - arr[index]) };
            }

        }

        return answer;

    }

    public static void main(String args[]) {

        int arr[] = { 3, 3 };
        int target = 6;

        TwoSum ts = new TwoSum();
        System.out.println(Arrays.toString(ts.twoSum(arr, target)));

    }

}
